extensions/route_map/util.rs

```diff
@@ -1,141 +1,51 @@
-use std::collections::HashMap;
+use std::borrow::Cow;
+use std::collections::HashSet;
 
-use pyo3::{
-    prelude::*,
-    types::{IntoPyDict, PyDict, PyTuple, PyType},
-};
+use pyo3::prelude::*;
 
-use crate::route_map::StarliteContext;
+use crate::wrappers::PathParameter;
 
-/// Splits a path on '/' and adds all of the 'components' to a new Vec.
-/// '/' itself is the first component by default since a leading slash
-/// is required by convention
-pub fn get_base_components(path: &str) -> Vec<&str> {
-    let path_split = path.split('/');
-
-    let mut components = Vec::<&str>::with_capacity(path_split.size_hint().0 + 1);
-    components.push("/");
-
-    path_split
-        .filter(|component| !component.is_empty())
-        .for_each(|component| components.push(component));
-
-    components
-}
-
-/// Given two HashMap slices representing lists of path parameter definition objects,
-/// returns whether they are equal. This is just implementing PartialEq, for this type,
-/// but with a normal function because a Python instance is required
-pub fn path_parameters_eq(
-    a: &[HashMap<String, Py<PyAny>>],
-    b: &[HashMap<String, Py<PyAny>>],
-    py: Python,
-) -> PyResult<bool> {
-    let mut eq = true;
-    if a.len() != b.len() {
-        eq = false;
-    } else {
-        for (a, b) in a.iter().zip(b.iter()) {
-            if !a.into_py_dict(py).eq(b.into_py_dict(py))? {
-                eq = false;
-                break;
-            }
-        }
-    }
-    Ok(eq)
+/// Iterator over components of a path (skipping empty components)
+pub(crate) fn get_base_components(path: &str) -> Vec<&str> {
```

Comment:
I don't agree with the reasoning behind that lint, and it [conflicts](https://github.com/rust-lang/rust-clippy/issues/5369) with an official rust lint (`unreachable_pub`).

I personally much prefer to know when reading the function "this is definitely not exposed outside this crate".

Reply:
Fair enough, I sometimes am also back and forth on this lint. I guess we are not exposing things from the crate in the normal sense anyway though, since we know only the things within #[pymethods] will be available to python and everything else is inaccessible. I'm not sure what we should do about visibility in general. I'm fine with leaving these pub(crate)s in the end